docs/dev/table/streaming/temporal_tables.md

```diff
@@ -57,134 +56,68 @@ SELECT * FROM product_changelog;
-operation execution time for every changelog, it's strongly recommended defines the database operation execution time 
-as event time rather than ingestion-time or time in the record, otherwise the version extract from the changelog may
-mismatch with the version in database.
- 
-### Defining Versioned View
+Flink also supports defining versioned views if the underlying query contains a unique key constraint and event-time attribute. Image an append-only table of currency rates. 
 
-Flink also supports defining versioned view only if the view contains unique key constraint and event time.
- 
-Let’s assume that we have the following table `RatesHistory`:
 {% highlight sql %}
--- Define an append-only table
-CREATE TABLE RatesHistory (
-    currency_time TIMESTAMP(3),
-    currency STRING,
-    rate DECIMAL(38, 10),
-    WATERMARK FOR currency_time AS currency_time   -- defines the event time
+CREATE TABLE currency_rates (
+	currency      STRING,
+	rate          DECIMAL(32, 10)
+	update_time   TIMESTAMP(3),
+	WATERMARK FOR update_time AS update_time
 ) WITH (
-  'connector' = 'kafka',
-  'topic' = 'rates',
-  'scan.startup.mode' = 'earliest-offset',
-  'properties.bootstrap.servers' = 'localhost:9092',
-  'format' = 'json'                                -- this is an append only source
-)
+	'connector' = 'kafka',
+	'topic'	    = 'rates',
+	'properties.bootstrap.servers' = 'localhost:9092',
+	'format'    = 'json'
+);
 {% endhighlight %}
 
-Table `RatesHistory` represents an ever growing append-only table of currency exchange rates with respect to 
-Yen (which has a rate of 1). For example, the exchange rate for the period from 09:00 to 10:45 of Euro to Yen was 114.
-From 10:45 to 11:15 it was 116.
-
-{% highlight sql %}
-SELECT * FROM RatesHistory;
-
-currency_time currency  rate
-============= ========= ====
-09:00:00      US Dollar 102
-09:00:00      Euro      114
-09:00:00      Yen       1
-10:45:00      Euro      116
-11:15:00      Euro      119
-11:49:00      Pounds    108
-{% endhighlight %}
+The table `currency_rates` contains a row for each currency - with respect to USD - and receives a new row each time the rate changes.
+The `JSON` format does not support native changelog semantics, and so Flink can only read this table as append-only.
+However, it is clear to us (the query developer) that this table has all the necessary information to define a versioned table. 
 
-To define a versioned table on `RatesHistory`, Flink supports defining a versioned view 
-by [deduplication query]({% link dev/table/sql/queries.md %}#deduplication) which produces an ordered changelog
-stream with an inferred primary key(`currency`) and event time(`currency_time`).
+Flink can reinterpret this table as a versioned table by defining a [deduplication query]({% link dev/table/sql/queries.md %}#deduplication) which produces an ordered changelog stream with an inferred primary key (currency) and event time (update_time). 
```

Comment:
The real answer is flink has a special optimization step that turns any `OVER WINDOW` with exactly this format into a versioned table keyed by the fields in the `PARTITION BY` clause. 

```sql
SELECT *,
      ROW_NUMBER() OVER (PARTITION BY [key,] ORDER BY [time attribute] DESC) AS row_num
WHERE 
     row_num = 1
```

In a batch query, this would return the most recent value for each currency. The streaming semantic is it provides the new value for each partition each time a more recent value is made available.